packages/inference/src/providers/providerHelper.ts

```diff
@@ -0,0 +1,137 @@
+import type { ChatCompletionOutput, TextGenerationOutput } from "@huggingface/tasks";
+import { InferenceOutputError } from "../lib/InferenceOutputError";
+import type { BodyParams, HeaderParams, UrlParams } from "../types";
+import { toArray } from "../utils/toArray";
+/**
+ * Base class for task-specific provider helpers
+ */
+export abstract class TaskProviderHelper {
+	constructor(
+		private provider: string,
+		private baseUrl: string,
+		private task?: string,
+		readonly clientSideRoutingOnly: boolean = false
+	) {}
+
+	/**
+	 * Return the response in the expected format.
+	 * Needs to be implemented in the subclasses.
+	 */
+	abstract getResponse(
+		response: unknown,
+		url?: string,
+		headers?: Record<string, string>,
+		outputType?: "url" | "blob"
+	): unknown;
+
+	/**
+	 * Prepare the base URL for the request
+	 */
+	makeBaseUrl(): string {
+		return this.baseUrl;
+	}
+
+	/**
+	 * Prepare the body for the request
+	 */
+	makeBody(params: BodyParams): unknown {
```

Comment:
This would avoid a cast by the caller, but requires to cast `params.args.data`

```suggestion
	makeBody(params: BodyParams): BodyInit {
```